### PR TITLE
[MC-292] Update AD/LDAP workflows

### PR DESCRIPTION
**Pull request by dsliwinski-r7**
## Requirements

Workflow contributors, verify you have completed the following items by checking them off:

### Checklist
- [ ] Workflow verified as working in a test
- [ ] Workflow bundle includes `Name_of_Workflow.icon` file, `help.md` file, `workflow.spec.yaml` file, and `screenshots` directory
- [ ] Workflow title should follow general [Title of Articles](http://grammar.yourdictionary.com/capitalization/rules-for-capitalization-in-titles.html) format
- [ ] Workflow includes a detailed Description
- [ ] Workflow step names follow [Title of Articles](http://grammar.yourdictionary.com/capitalization/rules-for-capitalization-in-titles.html) format
- [ ] Triggers, Decisions, and other workflow steps include descriptions where needed
- [ ] Workflow includes at least one `Final Report` artifact that summarizes workflow execution details
- [ ] Workflow bundle (directory) is named after the title of the workflow in snake case with first letters capitalized, e.g. `workflows/Name_of_Workflow/`
- [ ] `help.md` file includes Description, Key Features, Requirements, Setup, Technical Details, Version History, and References
- [ ] `workflow.spec.yaml` file includes `products`, `name`, `title`, `description`, `use_cases` tags, `keywords` tags, `source_url`, and the `name` for each screenshot matches a PNG image in the `screenshots` directory

### Assessment

Block Domain Account with Active Directory from InsightIDR
<details>

```
[*] Validating workflow at .

[*] Running Integration Validators...
[*] Executing validator WorkflowDirectoryNameMatchValidator
[*] Executing validator WorkflowFilesValidator
[*] Executing validator WorkflowHelpValidator
[*] Executing validator WorkflowChangelogValidator
[*] Executing validator WorkflowVendorValidator
[*] Executing validator WorkflowVersionValidator
[*] Executing validator WorkflowExtensionValidator
[*] Executing validator WorkflowSupportValidator
[*] Executing validator WorkflowPNGHashValidator
[*] Executing validator WorkflowICONFileNameValidator
[*] Executing validator WorkflowScreenshotValidator
[*] Executing validator WorkflowTitleValidator
[*] Executing validator WorkflowDescriptionValidator
[*] Executing validator WorkflowNameValidator
[*] Executing validator WorkflowProfanityValidator
[*] Executing validator WorkflowHelpPluginUtilizationValidator
[*] Executing validator WorkflowICONFileValidator
[*] Executing validator WorkflowEncodingValidator
[*] Executing validator WorkflowParametersKeywordValidator
[*] Executing validator UseCaseValidator
[*] Executing validator UnapprovedKeywordsValidator
[*] Executing validator WorkflowPythonScriptUseValidator
[*] Workflow successfully validated!

----
[*] Total time elapsed: 1299.102ms
```

<summary>
icon-validate .
</summary>
</details>

Disable a Domain User with Active Directory from an InsightIDR UBA Alert

<details>

```
[*] Validating workflow at .

[*] Running Integration Validators...
[*] Executing validator WorkflowDirectoryNameMatchValidator
[*] Executing validator WorkflowFilesValidator
[*] Executing validator WorkflowHelpValidator
[*] Executing validator WorkflowChangelogValidator
[*] Executing validator WorkflowVendorValidator
[*] Executing validator WorkflowVersionValidator
[*] Executing validator WorkflowExtensionValidator
[*] Executing validator WorkflowSupportValidator
[*] Executing validator WorkflowPNGHashValidator
[*] Executing validator WorkflowICONFileNameValidator
[*] Executing validator WorkflowScreenshotValidator
[*] Executing validator WorkflowTitleValidator
[*] Executing validator WorkflowDescriptionValidator
[*] Executing validator WorkflowNameValidator
[*] Executing validator WorkflowProfanityValidator
[*] Executing validator WorkflowHelpPluginUtilizationValidator
[*] Executing validator WorkflowICONFileValidator
[*] Executing validator WorkflowEncodingValidator
[*] Executing validator WorkflowParametersKeywordValidator
[*] Executing validator UseCaseValidator
[*] Executing validator UnapprovedKeywordsValidator
[*] Executing validator WorkflowPythonScriptUseValidator
[*] Workflow successfully validated!

----
[*] Total time elapsed: 1262.5349999999999ms
```

<summary>
icon-validate .
</summary>
</details>

Disable User in Active Directory from Microsoft Teams

<details>

```
[*] Validating workflow at .

[*] Running Integration Validators...
[*] Executing validator WorkflowDirectoryNameMatchValidator
[*] Executing validator WorkflowFilesValidator
[*] Executing validator WorkflowHelpValidator
[*] Executing validator WorkflowChangelogValidator
[*] Executing validator WorkflowVendorValidator
[*] Executing validator WorkflowVersionValidator
[*] Executing validator WorkflowExtensionValidator
[*] Executing validator WorkflowSupportValidator
[*] Executing validator WorkflowPNGHashValidator
[*] Executing validator WorkflowICONFileNameValidator
[*] Executing validator WorkflowScreenshotValidator
[*] Executing validator WorkflowTitleValidator
[*] Executing validator WorkflowDescriptionValidator
[*] Executing validator WorkflowNameValidator
[*] Executing validator WorkflowProfanityValidator
[*] Executing validator WorkflowHelpPluginUtilizationValidator
[*] Executing validator WorkflowICONFileValidator
[*] Executing validator WorkflowEncodingValidator
[*] Executing validator WorkflowParametersKeywordValidator
[*] Executing validator UseCaseValidator
[*] Executing validator UnapprovedKeywordsValidator
318
[*] Workflow successfully validated!

----
[*] Total time elapsed: 1192.0059999999999ms
```

<summary>
icon-validate .
</summary>
</details>

Force Password Reset in Active Directory from Microsoft Teams

<details>

```
[*] Validating workflow at .

[*] Running Integration Validators...
[*] Executing validator WorkflowDirectoryNameMatchValidator
[*] Executing validator WorkflowFilesValidator
[*] Executing validator WorkflowHelpValidator
[*] Executing validator WorkflowChangelogValidator
[*] Executing validator WorkflowVendorValidator
[*] Executing validator WorkflowVersionValidator
[*] Executing validator WorkflowExtensionValidator
[*] Executing validator WorkflowSupportValidator
[*] Executing validator WorkflowPNGHashValidator
[*] Executing validator WorkflowICONFileNameValidator
[*] Executing validator WorkflowScreenshotValidator
[*] Executing validator WorkflowTitleValidator
[*] Executing validator WorkflowDescriptionValidator
[*] Executing validator WorkflowNameValidator
[*] Executing validator WorkflowProfanityValidator
[*] Executing validator WorkflowHelpPluginUtilizationValidator
[*] Executing validator WorkflowICONFileValidator
[*] Executing validator WorkflowEncodingValidator
[*] Executing validator WorkflowParametersKeywordValidator
[*] Executing validator UseCaseValidator
[*] Executing validator UnapprovedKeywordsValidator
318
[*] Workflow successfully validated!

----
[*] Total time elapsed: 1290.097ms
```

<summary>
icon-validate .
</summary>
</details>

Force Password Reset in Active Directory from Slack

<details>

```
[*] Validating workflow at .

[*] Running Integration Validators...
[*] Executing validator WorkflowDirectoryNameMatchValidator
[*] Executing validator WorkflowFilesValidator
[*] Executing validator WorkflowHelpValidator
[*] Executing validator WorkflowChangelogValidator
[*] Executing validator WorkflowVendorValidator
[*] Executing validator WorkflowVersionValidator
[*] Executing validator WorkflowExtensionValidator
[*] Executing validator WorkflowSupportValidator
[*] Executing validator WorkflowPNGHashValidator
[*] Executing validator WorkflowICONFileNameValidator
[*] Executing validator WorkflowScreenshotValidator
[*] Executing validator WorkflowTitleValidator
[*] Executing validator WorkflowDescriptionValidator
[*] Executing validator WorkflowNameValidator
[*] Executing validator WorkflowProfanityValidator
[*] Executing validator WorkflowHelpPluginUtilizationValidator
[*] Executing validator WorkflowICONFileValidator
[*] Executing validator WorkflowEncodingValidator
[*] Executing validator WorkflowParametersKeywordValidator
[*] Executing validator UseCaseValidator
[*] Executing validator UnapprovedKeywordsValidator
318
[*] Workflow successfully validated!

----
[*] Total time elapsed: 1217.295ms
```

<summary>
icon-validate .
</summary>
</details>

InsightIDR Multi Country Auth and Remediation

<details>

```
[*] Validating workflow at .

[*] Running Integration Validators...
[*] Executing validator WorkflowDirectoryNameMatchValidator
[*] Executing validator WorkflowFilesValidator
[*] Executing validator WorkflowHelpValidator
[*] Executing validator WorkflowChangelogValidator
[*] Executing validator WorkflowVendorValidator
[*] Executing validator WorkflowVersionValidator
[*] Executing validator WorkflowExtensionValidator
[*] Executing validator WorkflowSupportValidator
[*] Executing validator WorkflowPNGHashValidator
[*] Executing validator WorkflowICONFileNameValidator
[*] Executing validator WorkflowScreenshotValidator
[*] Executing validator WorkflowTitleValidator
[*] Executing validator WorkflowDescriptionValidator
[*] Executing validator WorkflowNameValidator
[*] Executing validator WorkflowProfanityValidator
[*] Executing validator WorkflowHelpPluginUtilizationValidator
[*] Executing validator WorkflowICONFileValidator
[*] Executing validator WorkflowEncodingValidator
[*] Executing validator WorkflowParametersKeywordValidator
[*] Executing validator UseCaseValidator
[*] Executing validator UnapprovedKeywordsValidator
[*] Executing validator WorkflowPythonScriptUseValidator
	Python 3 Script use found in these steps
		[-] Step Name - Pyreplace
	Please review these steps carefully before use
[*] Workflow successfully validated!

----
[*] Total time elapsed: 1471.1879999999999ms
```

<summary>
icon-validate .
</summary>
</details>

Suspend AD LDAP User from Slack

<details>

```
[*] Validating workflow at .

[*] Running Integration Validators...
[*] Executing validator WorkflowDirectoryNameMatchValidator
[*] Executing validator WorkflowFilesValidator
[*] Executing validator WorkflowHelpValidator
[*] Executing validator WorkflowChangelogValidator
[*] Executing validator WorkflowVendorValidator
[*] Executing validator WorkflowVersionValidator
[*] Executing validator WorkflowExtensionValidator
[*] Executing validator WorkflowSupportValidator
[*] Executing validator WorkflowPNGHashValidator
[*] Executing validator WorkflowICONFileNameValidator
[*] Executing validator WorkflowScreenshotValidator
[*] Executing validator WorkflowTitleValidator
[*] Executing validator WorkflowDescriptionValidator
[*] Executing validator WorkflowNameValidator
[*] Executing validator WorkflowProfanityValidator
[*] Executing validator WorkflowHelpPluginUtilizationValidator
[*] Executing validator WorkflowICONFileValidator
[*] Executing validator WorkflowEncodingValidator
[*] Executing validator WorkflowParametersKeywordValidator
[*] Executing validator UseCaseValidator
[*] Executing validator UnapprovedKeywordsValidator
318
[*] Workflow successfully validated!

----
[*] Total time elapsed: 1235.261ms
```

<summary>
icon-validate .
</summary>
</details>

Tag Newly Discovered Devices with Active Directory OU from InsightVM

<details>

```
[*] Validating workflow at .

[*] Running Integration Validators...
[*] Executing validator WorkflowDirectoryNameMatchValidator
[*] Executing validator WorkflowFilesValidator
[*] Executing validator WorkflowHelpValidator
[*] Executing validator WorkflowChangelogValidator
[*] Executing validator WorkflowVendorValidator
[*] Executing validator WorkflowVersionValidator
[*] Executing validator WorkflowExtensionValidator
[*] Executing validator WorkflowSupportValidator
[*] Executing validator WorkflowPNGHashValidator
[*] Executing validator WorkflowICONFileNameValidator
[*] Executing validator WorkflowScreenshotValidator
[*] Executing validator WorkflowTitleValidator
[*] Executing validator WorkflowDescriptionValidator
[*] Executing validator WorkflowNameValidator
[*] Executing validator WorkflowProfanityValidator
[*] Executing validator WorkflowHelpPluginUtilizationValidator
[*] Executing validator WorkflowICONFileValidator
[*] Executing validator WorkflowEncodingValidator
[*] Executing validator WorkflowParametersKeywordValidator
[*] Executing validator UseCaseValidator
[*] Executing validator UnapprovedKeywordsValidator
[*] Executing validator WorkflowPythonScriptUseValidator
[*] Workflow successfully validated!

----
[*] Total time elapsed: 1592.817ms
```

<summary>
icon-validate .
</summary>
</details>

### Screenshots

<details>

block_domain_account_with_ad

block_domain_account_with_ad2

<summary>
Block Domain Account with Active Directory from InsightIDR
</summary>
</details>

<details>

disable_domain_user_with_ad

disable_domain_user_with_ad2

<summary>
Disable a Domain User with Active Directory from an InsightIDR UBA Alert
</summary>
</details>

<details>

disable_teams1

disable_teams2

disable_teams3

<summary>
Disable User in Active Directory from Microsoft Teams
</summary>
</details>

<details>

reset_teams1

reset_teams2

<summary>
Force Password Reset in Active Directory from Microsoft Teams
</summary>
</details>

<details>

reset_slack1

reset_slack2

<summary>
Force Password Reset in Active Directory from Slack
</summary>
</details>

<details>

insightidr_multi_country_auth2

InsightIDR_multi_country_auth

<summary>
InsightIDR Multi Country Auth and Remediation
</summary>
</details>

<details>

disable_slack1

disable_slack2

disable_slack3

<summary>
Suspend AD LDAP User from Slack
</summary>
</details>

<details>

tag

tag2

tag3

<summary>
Tag Newly Discovered Devices with Active Directory OU from InsightVM
</summary>
</details>

### CODE REVIEW

File: workflows/Disable_User_in_Active_Directory_from_Microsoft_Teams/help.md [hunk 1]
```diff
@@ -26,9 +26,17 @@ Sample Microsoft Teams Trigger Commands:
...
 
-Once the workflow has been imported, **The `Find DN` step and each Microsoft Teams step will need to be updated to suit your Microsoft Teams and LDAP environments.** Edit the input with the preset text of `change_me` in each of these steps in the workflow.
+This workflow leverages InsightConnect's Parameters feature. This feature allows variables used multiple times throughout a workflow to be entered once and then referenced throughout the workflow.
 
-After configuring the Teams steps, activate the workflow in order to trigger it.
+There are three parameters you will need to configure in order to complete setup of your workflow:
+
+* Team Name: The Microsoft Teams team name in your environment where the workflow should be triggered and respond
+* Channel Name: The Microsoft Teams channel name in your environment where the workflow should be triggered and respond (the channel should exist in the aforementioned team)
+* Search Base: The base of the Active Directory user search request, eg. DC=example,DC=com
```
Comment (pmara-r7) [inline workflows/Disable_User_in_Active_Directory_from_Microsoft_Teams/help.md:35]:
```suggestion
* Search Base: The base of the Active Directory user search request, e.g. DC=example,DC=com
```

File: workflows/InsightIDR_MultiCountry_Auth_and_Remediation/InsightIDR_MultiCountry_Auth_and_Remediation.icon [hunk 2]
```diff
@@ -581,7 +584,7 @@
             "defaultInputJSONSchema": {
               "properties": {
                 "distinguished_name": {
-                  "description": "The distinguished name of the user to disable e.g. CN=user,OU=domain_users,DC=mydomain,DC=com",
+                  "description": "The distinguished name of the user to disable",
```
Comment (pmara-r7) [inline workflows/InsightIDR_MultiCountry_Auth_and_Remediation/InsightIDR_MultiCountry_Auth_and_Remediation.icon:587]:
```suggestion
                  "description": "The distinguished name of the user to disable, e.g. CN=user,OU=domain_users,DC=mydomain,DC=com",
```

File: workflows/InsightIDR_MultiCountry_Auth_and_Remediation/InsightIDR_MultiCountry_Auth_and_Remediation.icon [hunk 3]
```diff
@@ -1210,36 +1213,47 @@
...
+                },
                 "search_base": {
                   "description": "The base of the search request",
                   "order": 2,
                   "title": "Search Base",
                   "type": "string"
                 },
                 "search_filter": {
-                  "description": "The filter of the search request. It must conform to the LDAP filter syntax specified in RFC4515. Example: (accountName=joesmith)",
+                  "description": "The filter of the search request. It must conform to the LDAP filter syntax specified in RFC4515",
```
Comment (pmara-r7) [inline workflows/InsightIDR_MultiCountry_Auth_and_Remediation/InsightIDR_MultiCountry_Auth_and_Remediation.icon:1248]:
```suggestion
                  "description": "The filter of the search request. It must conform to the LDAP filter syntax specified in RFC4515. Example: (accountName=joesmith)",
```

**Review by pmara-r7 — CHANGES_REQUESTED** [review]

**Comment by pmara-r7** [comment]
Just two small comments: adding back in two examples. Feel free to accept or reject the two examples.

**Comment by dsliwinski-r7** [comment]
> Just two small comments: adding back in two examples. Feel free to accept or reject the two examples.

@pmara-r7 We shouldn't change these descriptions because they have been changed after updating the plugin version in the workflow. If we want to add these examples back then I think we should do it directly in the plugin :)

**Review by pmara-r7 — APPROVED** [review]

**Review by mberezin-r7 — APPROVED** [review]

**Review by mrinehart-r7 — APPROVED** [review]

**Pull request merged**